core/src/main/java/org/moditect/commands/GenerateModuleInfo.java

```diff
@@ -297,28 +297,6 @@ private ModuleDeclaration applyOpensPatterns(ModuleDeclaration moduleDeclaration
         return moduleDeclaration;
     }
 
-    private ClassOrInterfaceType getType(String fqn) {
-        String[] parts = fqn.split( "\\." );
-
-        ClassOrInterfaceType scope = null;
-        String name = null;
-
-        if ( parts.length == 1 ) {
-            scope = null;
-            name = parts[0];
-        }
-        else {
-            ClassOrInterfaceType parentScope = null;
-            for( int i = 0; i < parts.length - 1; i++ ) {
-                scope = new ClassOrInterfaceType( parentScope, parts[i] );
-                parentScope = scope;
-            }
-            name = parts[parts.length - 1];
-        }
-
-        return new ClassOrInterfaceType( scope, name );
-    }
```

Comment:
Hurray! Someone noted that the grammar needs only a Name, so we can get away with a simple `parseName` instead of this tower of logic.

Reply:
Nice :)